<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Coursework 1</title>

<script src="coursework_1_files/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="coursework_1_files/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="coursework_1_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="coursework_1_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="coursework_1_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="coursework_1_files/navigation-1.1/tabsets.js"></script>
<link href="coursework_1_files/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="coursework_1_files/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->





<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Coursework 1</h1>

</div>


<div id="jennifer-stiens" class="section level2">
<h2>Jennifer Stiens</h2>
<p><a href="mailto:j.j.stiens@gmail.com">j.j.stiens@gmail.com</a></p>
<p><a href="https://github.com/jenjane118">github.com/jenjane118</a></p>
</div>
<div id="question-1" class="section level2">
<h2>Question 1</h2>
<p><em>Using good NGS practices, remap Negative.fq so that the mapping statistics improve.</em></p>
<ol style="list-style-type: lower-alpha">
<li>View file and work out what is wrong with reads. Generate fastqc report.</li>
</ol>
<p>First I will generate a Fastqc report of the trimmed_Negative.fq sequence file.</p>
<pre class="bash"><code>/s/software/fastqc/v0.11.8/FastQC/fastqc trimmed_Negative.fq
</code></pre>
<!-- ![sequence quality plot](/d/projects/u/sj003/course_materials/fastq/coursework_1/trimmed_Negative_fastqc.png) -->
<div class="figure">
<img src="trimmed_Negative_fastqc.png" alt="sequence quality plot" />
<p class="caption">sequence quality plot</p>
</div>
<p><a href="/d/projects/u/sj003/course_materials/fastq/coursework_1/trimmed_Negative_fastqc.html">fastqc report original</a></p>
<p>This shows a dropoff of sequence quality at the ends of each read. Looking at the fasta file on the shell screen, it is obvious that the reads have 5 N’s at the 5’ end and 4 N’s at the 3’ end of each read that need to be removed. To do this, we use cutadapt to process each read.</p>
<pre class="bash"><code>/s/software/anaconda/python3/bin/cutadapt --trim-n -o trimmedNs_negative.fq trimmed_Negative.fq
less trimmedNs_negative.fq
</code></pre>
<p>Running Fastqc again, you can see an improvement in the sequence quality.</p>
<pre class="bash"><code>/s/software/fastqc/v0.11.8/FastQC/fastqc trimmedNs_Negative.fq</code></pre>
<!-- ![sequence quality plot](/d/projects/u/sj003/course_materials/fastq/coursework_1/trimmedNs_negative_fastqc.png) -->
<div class="figure">
<img src="trimmedNs_negative_fastqc.png" alt="sequence quality plot" />
<p class="caption">sequence quality plot</p>
</div>
<p><a href="/d/projects/u/sj003/course_materials/fastq/coursework_1/trimmedNs_Negative_fastqc.html">fastqc report trimmed</a></p>
<ol start="2" style="list-style-type: lower-alpha">
<li>Use bowtie2 to align the reads to reference genome using two different options.</li>
</ol>
<p>To align both the original file and the trimmed file using bowtie2 against the reference genome, AFPN02.1, and generate a .sam file to use for further analysis:</p>
<pre class="bash"><code>time /s/software/anaconda/python3/bin/bowtie2 --end-to-end -x ${st_path}/course_materials/genomes/AFPN02.1/AFPN02.1_merge -q ${st_path}/course_materials/fastq/trimmed_Negative.fq -S Negative.sam 2&gt; Negative_bowtie_stats.txt</code></pre>
<pre class="bash"><code>time /s/software/anaconda/python3/bin/bowtie2 --end-to-end -x ${st_path}/course_materials/genomes/AFPN02.1/AFPN02.1_merge -q ${st_path}/course_materials/fastq/trimmedNs_Negative.fq -S Negative2.sam 2&gt; Negative2_bowtie_stats.txt</code></pre>
<p>The text file can be examined on the bash terminal, and the .sam file can be analysed in a multiqc display showing that the new Negative alignment (Negative2) has the same stats as Positive:</p>
<pre class="bash"><code>/s/software/anaconda/python3/bin/multiqc . -f</code></pre>
<!-- ![multiq alignment scores](/d/projects/u/sj003/results_cw1/Neg_v_Pos_multiq.png) -->
<div class="figure">
<img src="Neg_v_Pos_multiq.png" alt="multiq alignment scores" />
<p class="caption">multiq alignment scores</p>
</div>
<p>Another option is to align the Negative file using options in Bowtie2 that ignore the first and last bases when aligning:</p>
<pre class="bash"><code>time /s/software/anaconda/python3/bin/bowtie2 --end-to-end --trim5 5 --trim3 4 -x ${st_path}/course_materials/genomes/AFPN02.1/AFPN02.1_merge -q ${st_path}/course_materials/fastq/trimmed_Negative.fq -S Negative3.sam 2&gt; Negative3_bowtie_stats.txt</code></pre>
<div class="figure">
<img src="cw1_q1_multiqc.png" alt="multiqc sequence stats" />
<p class="caption">multiqc sequence stats</p>
</div>
<p>This gives the same alignment results as using cutadapt.</p>
<p>I can also try a local alignment on the file where I cut the Ns off to see if this improves the alignment:</p>
<pre class="bash"><code>time /s/software/anaconda/python3/bin/bowtie2 --local -x ${st_path}/course_materials/genomes/AFPN02.1/AFPN02.1_merge -q ${st_path}/course_materials/fastq/trimmedNs_Negative.fq -S Negative4.sam 2&gt; Negative4_bowtie_stats.txt</code></pre>
<ol start="3" style="list-style-type: lower-alpha">
<li>samtools stats:</li>
</ol>
<p>Negative.sam original</p>
<p>Negative2.sam trimmed Ns with cutadapt</p>
<p>Negative3.sam aligned with trimming Ns with bowtie</p>
<p>Negative4.sam local alignment</p>
<p>In order to save time running samtools for each file, I wrote a bash script (d/projects/u/sj003/results_cw1/samtools_bash.sh)</p>
<pre class="bash"><code>#!/bin/bash
# samtools_bash.sh
# script to run basic samtools functions for each file in folder:

# Assign everything ending in .sam to SAMFILES variable
SAMFILES=*.sam

# Loop over SAMFILES and run samtools for each file
for file in $SAMFILES
do
  filename=$(basename &quot;$file&quot;)
  extension=&quot;${filename##*.}&quot;
  filename=&quot;${filename%.*}&quot;
  
  echo &quot;Sort file:        $file&quot;
  echo &quot;Index file:       $filename.bam&quot;
  echo &quot;Create stats for file:        $file&quot;
  echo &quot;Create flagstat for file:       $file&quot;

  #Call samtools functions for each file
  /s/software/samtools/v1.9/bin/samtools sort ${file} &gt; ${filename}.bam
  /s/software/samtools/v1.9/bin/samtools index ${filename}.bam
  /s/software/samtools/v1.9/bin/samtools stats ${file} &gt; ${filename}_stats.txt
  /s/software/samtools/v1.9/bin/samtools flagstat ${file} &gt; ${filename}_flagstat.txt

  echo -e &quot;######################\n\n&quot;
done
</code></pre>
<p>Call the script using:</p>
<pre class="bash"><code>bash samtools_bash.sh
</code></pre>
<p>Run a MultiQC report to summarise the sequence and alignment data for all the alignments.</p>
<pre class="bash"><code>/s/software/anaconda/python3/bin/multiqc . -n q1_multiqc_report</code></pre>
<div class="figure">
<img src="MultiQCstats_Q1.png" alt="multiqc sequence stats" />
<p class="caption">multiqc sequence stats</p>
</div>
<p><a href="/d/projects/u/sj003/results_cw1/q1_multiqc_report.html">multiq report</a></p>
<p>The report shows that the best quality alignment is for the local alignment of the sequences that had the 5’ and 3’ N’s cut off using cutadapt (Negative4). This alignment had a slightly better error rate (0.10%) than the Positive alignment (which was done using the end-to-end alignmnent), or the edited Negative sequence using the end-to-end alignment. The local alignment allows some of the bases on the ends to be omitted to get a better alignment score. It basically just skips a base at the beginning or end if it doesn’t align, and that may account for the very slightly better error rater. All 1.1M reads were mapped. I think it is a better idea to edit the N’s out of the sequence reads at the outset, using cutadapt, rather than to just ignore them using the settings in bowtie2. This way the fastqc reports will accurately represent the overall sequence quality and the sequence reads can be used in other software applications without further adjustment.</p>
</div>
<div id="question-2" class="section level2">
<h2>Question 2</h2>
<p><em>Use cutadapt to re-process the BQ.fq file and bowtie2 to map reads to reference genome. Discuss trade-off between improved mapping rate and error rate. Which is more important considering goal of coming up with most accurate genomic reference sequence for sampled bacterium?</em></p>
<p>The first step is to examine the fasta file for the BQ sequence reads. Similar to Negative sequences above, they contain 5’ and 3’ Ns that need to be trimmed before alignment can occur. Using cutadapt, this can be accomplished the same way as we did in question 1:</p>
<pre class="bash"><code>/s/software/anaconda/python3/bin/cutadapt --trim-n -o trimmedns_BQ.fq trimmed_BQ.fq
less trimmedNs_negative.fq</code></pre>
<p>We can examine the fastqc report of the edited sequence:</p>
<pre class="bash"><code>/s/software/fastqc/v0.11.8/FastQC/fastqc trimmedns_BQ.fq</code></pre>
<div class="figure">
<img src="trimmedns_BQ_fastqc.png" alt="fastqc sequence quality" />
<p class="caption">fastqc sequence quality</p>
</div>
<p>This shows that though the quality of the ends is no longer as poor, the entire sequence has rather low sequence quality, with an average Phred score of 21.</p>
<p>Looking at the alignment of BQ with the bacterial genome reference sequence, we can see the mapping statistics are rather poor at an overall alignment rate of 82.82%:</p>
<pre class="bash"><code>time /s/software/anaconda/python3/bin/bowtie2 --end-to-end -x ${st_path}/course_materials/genomes/AFPN02.1/AFPN02.1_merge -q ${st_path}/course_materials/fastq/trimmedns_BQ.fq -S trimmednsBQ.sam 2&gt; trimmednsBQ_bowtie_stats.txt
less trimmednsBQ_bowtie_stats.txt</code></pre>
<p>I also performed bowtie2 alignment with the ‘very sensitive local’ setting to increase the alignment sensitivity and hopefully reduce the error rate (using local alignment) at the same time, which brings the overall alignment rate up to 89.43%.</p>
<pre class="bash"><code>time /s/software/anaconda/python3/bin/bowtie2 --very-sensitive-local -x ${st_path}/course_materials/genomes/AFPN02.1/AFPN02.1_merge -q ${st_path}/course_materials/fastq/trimmedns_BQ.fq -S trimmednsBQ_vsl.sam 2&gt; trimmednsBQvsl_bowtie_stats.txt
less trimmednsBQvsl_bowtie_stats.txt</code></pre>
<p>The ‘very sensitive’ setting has preset parameters that are designed to maximise sensitivity and accuracy. However, you can manually adjust these parameters, for example changing the the number of mismatches permitted per ‘seed’ in the alignment (using -N). I tried changing this parameter to 0, but using the same settings for the rest of the parameters as the very-sensitive-local setting uses (-D 20 -R 3 -N 1 -L 20 -i S,1,0.50). This increases the overall alignment rate to 98.64% but at what cost to the error rate?</p>
<pre class="bash"><code>time /s/software/anaconda/python3/bin/bowtie2 --local -D 20 -R 3 -N 1 -L 20 -i S,1,0.50 -x ${st_path}/course_materials/genomes/AFPN02.1/AFPN02.1_merge -q ${st_path}/course_materials/fastq/trimmedns_BQ.fq -S trimmednsBQ_cust.sam 2&gt; trimmednsBQcust_bowtie_stats.txt
less trimmednsBQcust_bowtie_stats.txt</code></pre>
<p>To evaluate the statistics for the alignments, I will call the samtools script to run the samtools functions for these files.</p>
<pre class="bash"><code>bash samtools_bash.sh
</code></pre>
<p>Run a MultiQC report to summarise the sequence and alignment data for all the alignments.</p>
<pre class="bash"><code>/s/software/anaconda/python3/bin/multiqc . -f -n q2_multiqc_report</code></pre>
<div class="figure">
<img src="Multiqc_report_q2.png" alt="Multiqc report" />
<p class="caption">Multiqc report</p>
</div>
<p><a href="/d/projects/u/sj003/results_cw1/q2_multiqc_report.html">multiqc report</a></p>
<table>
<thead>
<tr class="header">
<th>Sequence</th>
<th>MReads mapped/%</th>
<th>Error Rate</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>trimmednsBQ</td>
<td>0.9 / 82.8%</td>
<td>4.63%</td>
</tr>
<tr class="even">
<td>trimmednsBQ_vsl</td>
<td>1.0 / 89.4%</td>
<td>4.15%</td>
</tr>
<tr class="odd">
<td>trimmednsBQ_cust</td>
<td>1.1 / 98.6%</td>
<td>4.51%</td>
</tr>
</tbody>
</table>
<p>The report shows that there is a trade-off between alignment rate and error rate. To achieve the best alignment percentage, we have to accept an error rate of 4.51% using the custom parameters (trimmednsBQ_cust). The next best alignment rate, using the very sensitive local parameters (trimmednsBQ_vsl) has a lower error rate of 4.15%. While this is better, I am not sure if an error rate above 4% is acceptable for a consensus genome. Perhaps the sequencing runs should be repeated to try to get more high quality sequences.</p>
</div>
<div id="question-3" class="section level2">
<h2>Question 3</h2>
<p><em>Split the BQ aligned files into SAM files containing subsets of the full set of aligned reads.</em></p>
<p><em>Split into:</em></p>
<p><em>a) one file containing multimapping reads and another file with only uniquely mapped reads</em></p>
<pre class="bash"><code># for uniquely mapped reads (with MAPQ &lt; 10); all other files in multiBQ file (-U)
/s/software/samtools/v1.9/bin/samtools view -bq 10 trimmednsBQ.bam &gt; uniqueBQ.bam -U multiBQ.bam</code></pre>
<p><em>b) one file with unmapped reads and one with only mapped reads</em></p>
<pre class="bash"><code># includes unique reads (removes unmapped reads with flag of [4])
/s/software/samtools/v1.9/bin/samtools view -bF 4 trimmednsBQ.bam &gt; mappedBQ.bam

# includes only unmapped reads (with flag [4])
/s/software/samtools/v1.9/bin/samtools view -bf 4 trimmednsBQ.bam &gt; unmappedBQ.bam</code></pre>
<p><em>c)How can output from b be obtained using Bowtie2 instead?</em></p>
<p>Using bowtie2 -k option allows us to report the sequences with the desired number of alignments, which in this case I am using 1 to get reads that have only matched once. Unfortunately, this leaves the unaligned sequences in the file. Looking for solutions online, it seems there was a different command in bowtie (versus bowtie2), -m, that was able to filter by number of hits. Looking online, all the advice seems to be that the most logical thing is to do this using the flags in samtools, like I’ve done in b.</p>
<pre class="bash"><code>## filters for only unique reads (-k) but still contains unaligned seqs
time /s/software/anaconda/python3/bin/bowtie2 --end-to-end -k 1 -x ${st_path}/course_materials/genomes/AFPN02.1/AFPN02.1_merge -q ${st_path}/course_materials/fastq/trimmedns_BQ.fq -S trimmednsBQ2.sam 2&gt; trimmednsBQ2_stats.txt
</code></pre>
<p><em>d) Use BLAST to identify origin of unmapped reads.</em></p>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
